File: backend/src/post/post.dto.ts
import { ApiProperty } from "@nestjs/swagger";

export class FindAllPosts {
  @ApiProperty({ required: false })
  status: boolean;

  @ApiProperty({ required: false })
  offset: number;

  @ApiProperty({ required: false })
  limit: number;

  @ApiProperty({
    required: false,
    description: "Whether to include brand relation",
  })
  brand: string;

  @ApiProperty({
    required: false,
    description: "Whether to include model relation",
  })
  model: string;

  @ApiProperty({ required: false })
  brandFilter: string;

  @ApiProperty({ required: false })
  modelFilter: string;

  @ApiProperty({
    required: false,
    description: "Whether to include photo relation",
  })
  photo: string;

  @ApiProperty({
    required: false,
    description: "Whether to include category relation",
  })
  category: string;

  @ApiProperty({ required: false })
  categoryFilter: string;

  @ApiProperty({
    required: false,
    description: "Whether to include subscription relation",
  })
  subscription: string;

  @ApiProperty({ required: false, enum: ["asc", "desc"] })
  sortAs: string;

  @ApiProperty({
    required: false,
    enum: ["createdAt", "price", "year", "milleage"],
  })
  sortBy: string;

  @ApiProperty({ required: false })
  search: string;

  @ApiProperty({ required: false })
  minYear: string;

  @ApiProperty({ required: false })
  maxYear: string;

  @ApiProperty({ required: false })
  minPrice: string;

  @ApiProperty({ required: false })
  maxPrice: string;

  @ApiProperty({ required: false })
  transmission: string;

  @ApiProperty({ required: false })
  engineType: string;

  @ApiProperty({ required: false })
  enginePower: number;

  @ApiProperty({ required: false })
  milleage: string;

  @ApiProperty({ required: false })
  condition: string;

  @ApiProperty({ required: false })
  region: string;

  @ApiProperty({ required: false })
  location: string;

  @ApiProperty({ required: false })
  credit: string;

  @ApiProperty({ required: false })
  exchange: string;

  @ApiProperty({ required: false })
  subFilter: string | string[];

  @ApiProperty({ required: false })
  color: string;
}
export class listPost {
  @ApiProperty()
  uuids: string[];
  @ApiProperty({ required: false })
  brand: string;
  @ApiProperty({ required: false })
  model: string;

  @ApiProperty({ required: false })
  photo: string;
}

export class FindOnePost {
  @ApiProperty({ required: false })
  brand: string;
  @ApiProperty({ required: false })
  comment: string;
  @ApiProperty({ required: false })
  model: string;
  @ApiProperty({ required: false })
  subscription: string;
  @ApiProperty({ required: false })
  photo: string;
}
export class FindOneUUID {
  @ApiProperty()
  uuid: string;
}

export class CreatePost {
  @ApiProperty({})
  brandsId: string;
  @ApiProperty()
  location: string;
  @ApiProperty()
  modelsId: string;
  @ApiProperty()
  phone: string;
  @ApiProperty()
  condition: string;
  @ApiProperty()
  transmission: string;
  @ApiProperty()
  engineType: string;
  @ApiProperty()
  enginePower: number;
  @ApiProperty()
  year: number;
  @ApiProperty()
  credit: boolean;
  @ApiProperty()
  exchange: boolean;
  @ApiProperty()
  milleage: number;
  @ApiProperty()
  vin: string;
  @ApiProperty()
  price: number;
  @ApiProperty()
  currency: string;
  @ApiProperty()
  personalInfo: { name: string; location: string; region?: string } | null;
  @ApiProperty()
  description: string;
  @ApiProperty()
  subscriptionId: string;

  @ApiProperty({ required: false })
  color: string;
}

export class UpdatePost {
  @ApiProperty()
  brandsId: string;
  @ApiProperty()
  modelsId: string;
  @ApiProperty()
  condition: string;
  @ApiProperty()
  transmission: string;
  @ApiProperty()
  engineType: string;
  @ApiProperty()
  enginePower: number;
  @ApiProperty()
  year: number;
  @ApiProperty()
  milleage: number;
  @ApiProperty()
  vin: string;
  @ApiProperty()
  credit: boolean;
  @ApiProperty()
  exchange: boolean;
  @ApiProperty()
  price: number;
  @ApiProperty()
  currency: string;
  @ApiProperty()
  personalInfo: { name: string; location: string; region?: string } | null;
  @ApiProperty()
  description: string;
  @ApiProperty()
  subscriptionId: string;
  @ApiProperty()
  status: boolean;
  @ApiProperty({ required: false })
  color: string;
}

File: auto.tm-main/lib/data/dtos/post_dto.dart
import 'package:get/get.dart';


/// Post data transfer object - reflects the raw JSON from backend
class PostDto {
  final String uuid;
  final String? brandName;
  final String? modelName;
  final String? brandsId;
  final String? modelsId;
  final num? price;
  final num? year;
  final num? milleage;
  final String? engineType;
  final num? enginePower;
  final String? transmission;
  final String? condition;
  final String? vin;
  final String? currency;
  final String? description;
  final String? location;
  final bool? status;
  final bool? exchange;
  final bool? credit;
  final Map<String, dynamic>? personalInfo;
  final List<dynamic>? photo;
  final Map<String, dynamic>? subscription;
  final Map<String, dynamic>? video;
  final Map<String, dynamic>? file;
  final String? createdAt;

  PostDto({
    required this.uuid,
    this.brandName,
    this.modelName,
    this.brandsId,
    this.modelsId,
    this.price,
    this.year,
    this.milleage,
    this.engineType,
    this.enginePower,
    this.transmission,
    this.condition,
    this.vin,
    this.currency,
    this.description,
    this.location,
    this.status,
    this.exchange,
    this.credit,
    this.personalInfo,
    this.photo,
    this.subscription,
    this.video,
    this.file,
    this.createdAt,
  });

  factory PostDto.fromJson(Map<String, dynamic> json) {
    return PostDto(
      uuid: json['uuid']?.toString() ?? json['id']?.toString() ?? '',
      brandName: json['brandName']?.toString(),
      modelName: json['modelName']?.toString(),
      brandsId: json['brandsId']?.toString(),
      modelsId: json['modelsId']?.toString(),
      price: json['price'] as num?,
      year: json['year'] as num?,
      milleage: json['milleage'] as num?,
      engineType: json['engineType']?.toString(),
      enginePower: json['enginePower'] as num?,
      transmission: json['transmission']?.toString(),
      condition: json['condition']?.toString(),
      vin: json['vin']?.toString(),
      currency: json['currency']?.toString(),
      description: json['description']?.toString(),
      location: json['location']?.toString(),
      status: json['status'] as bool?,
      exchange: json['exchange'] as bool?,
      credit: json['credit'] as bool?,
      personalInfo: json['personalInfo'] as Map<String, dynamic>?,
      photo: json['photo'] as List<dynamic>?,
      subscription: json['subscription'] as Map<String, dynamic>?,
      video: json['video'] as Map<String, dynamic>?,
      file: json['file'] as Map<String, dynamic>?,
      createdAt: json['createdAt']?.toString(),
    );
  }
}
